src/tr2/game/effects.c

```diff
@@ -81,42 +104,42 @@ int16_t __cdecl Effect_Create(const int16_t room_num)
 
 void __cdecl Effect_Kill(const int16_t effect_num)
 {
-    EFFECT *const effect = &g_Effects[effect_num];
+    EFFECT *const effect = Effect_Get(effect_num);
     M_RemoveActive(effect_num);
     M_RemoveDrawn(effect_num);
 
-    effect->next_free = g_NextEffectFree;
-    g_NextEffectFree = effect_num;
+    effect->next_free = m_NextEffectFree;
+    m_NextEffectFree = effect_num;
 }
 
 void __cdecl Effect_NewRoom(const int16_t effect_num, const int16_t room_num)
 {
-    EFFECT *const effect = &g_Effects[effect_num];
-    ROOM *room = &g_Rooms[effect->room_num];
+    EFFECT *const effect = Effect_Get(effect_num);
+    ROOM *room = Room_Get(effect->room_num);
 
     int16_t link_num = room->effect_num;
     if (link_num == effect_num) {
         room->effect_num = effect->next_free;
     } else {
-        while (link_num != NO_ITEM) {
-            if (g_Effects[link_num].next_free == effect_num) {
-                g_Effects[link_num].next_free = effect->next_free;
+        while (link_num != NO_EFFECT) {
+            if (m_Effects[link_num].next_free == effect_num) {
+                m_Effects[link_num].next_free = effect->next_free;
                 break;
             }
-            link_num = g_Effects[link_num].next_free;
+            link_num = m_Effects[link_num].next_free;
         }
     }
 
+    room = Room_Get(effect->room_num);
```

Comment:
I think this should be `Room_Get(room_num);`.

Reply:
Thank you – that was it.